lib/postgres_test.go

```diff
@@ -1,10 +1,284 @@
 package lib
 
 import (
+	"context"
+	"fmt"
+	"github.com/deso-protocol/core/migrate"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/network"
+	"github.com/docker/docker/client"
+	"github.com/docker/go-connections/nat"
+	"github.com/go-pg/pg/v10"
+	migrations "github.com/robinjoseph08/go-pg-migrations/v3"
 	"github.com/stretchr/testify/require"
+	"io/ioutil"
+	"strconv"
+	"strings"
 	"testing"
+	"time"
 )
 
+// Set to true if you don't want to use the postgres testing framework.
+const DISABLE_POSTGRES_DOCKER_API = false
+
+// Config variables for spawning the postgres container.
+const CONTAINER_NAME = "test_postgresql"
+const LOCAL_PORT = "5433"
+const REMOTE_PORT = "5432"
+const POSTGRES_USER = "postgres"
+const POSTGRES_PASSWORD = "postgres"
```

Comment:
I also think we lose a lot by defining the postgres params in consts like this and dropping support for the PG_URI env var which is a common way to pass in pg params. The idea may be to encapsulate all postgres logic away from the user so they don't even need to know which params are being used, but what if for example, I'm already running a secondary postgres docker container on port 5433 and need to be able to configure the port.